src/qt/metadexdialog.cpp

```diff
@@ -107,21 +107,78 @@ void MetaDExDialog::setClientModel(ClientModel *model)
     this->clientModel = model;
     if (NULL != model) {
         connect(model, SIGNAL(refreshOmniState()), this, SLOT(OrderRefresh()));
-        connect(model, SIGNAL(refreshOmniBalance()), this, SLOT(OrderRefresh()));
+        connect(model, SIGNAL(refreshOmniBalance()), this, SLOT(BalanceOrderRefresh()));
         connect(model, SIGNAL(reinitOmniState()), this, SLOT(FullRefresh()));
     }
 }
 
 void MetaDExDialog::setWalletModel(WalletModel *model)
 {
+    // use wallet model to get visibility into BTC balance changes for fees
     this->walletModel = model;
-    if (model != NULL) { } // do nothing, signals from walletModel no longer needed
+    if (model != NULL) {
+       connect(model, SIGNAL(balanceChanged(CAmount,CAmount,CAmount,CAmount,CAmount,CAmount)), this, SLOT(UpdateBalances()));
+    }
+}
+
+void MetaDExDialog::PopulateAddresses()
+{
+    LOCK(cs_tally);
+
+    uint32_t propertyId = global_metadex_market;
+    bool testeco = false;
+    if (propertyId >= TEST_ECO_PROPERTY_1) testeco = true;
+
+    // get currently selected addresses
+    QString currentSetBuyAddress = ui->buyAddressCombo->currentText();
+    QString currentSetSellAddress = ui->sellAddressCombo->currentText();
+
+    // clear address selectors
+    ui->buyAddressCombo->clear();
+    ui->sellAddressCombo->clear();
+
+    // populate buy and sell addresses
+    for (std::map<string, CMPTally>::iterator my_it = mp_tally_map.begin(); my_it != mp_tally_map.end(); ++my_it) {
+        string address = (my_it->first).c_str();
+        unsigned int id;
+        (my_it->second).init();
+        while (0 != (id = (my_it->second).next())) {
+            if (id == propertyId) {
+                if (!getUserAvailableMPbalance(address, propertyId)) continue; // ignore this address, has no available balance to spend
+                if (IsMyAddress(address) == ISMINE_SPENDABLE) ui->sellAddressCombo->addItem((my_it->first).c_str()); // only include wallet addresses
+            }
+            if (id == OMNI_PROPERTY_MSC && !testeco) {
+                if (!getUserAvailableMPbalance(address, OMNI_PROPERTY_MSC)) continue;
+                if (IsMyAddress(address) == ISMINE_SPENDABLE) ui->buyAddressCombo->addItem((my_it->first).c_str());
+            }
+            if (id == OMNI_PROPERTY_TMSC && testeco) {
+                if (!getUserAvailableMPbalance(address, OMNI_PROPERTY_TMSC)) continue;
+                if (IsMyAddress(address) == ISMINE_SPENDABLE) ui->buyAddressCombo->addItem((my_it->first).c_str());
+            }
+        }
+    }
+
```

Comment:
Ahhh, I was wondering why you'd enclosed the previous address population code in ```{ }```, I thought that was a no-op but now it makes complete sense.

Thanks dude! :)  Confirmed working.